src/ledger/TrustLineWrapper.cpp

```diff
@@ -377,6 +391,13 @@ TrustLineWrapper::IssuerImpl::isAuthorizedToMaintainLiabilities() const
     return true;
 }
 
+bool
+TrustLineWrapper::IssuerImpl::isClawbackEnabled() const
+{
+    // an issuer can't clawback from itself
```

Comment:
I'm not sure about this. An issuer definitely shouldn't be able to do a normal clawback on itself (because that has no effect). But clawing back an issuer claimable balance is observable, so maybe that should work? @leighmcculloch thoughts?

Reply:
I agree with the comment, an issuer account can't clawback from itself. If the issuer account creates a claimable balance containing their asset, they can claw it back, the claimable balance is not the issuer account.

Reply:
After talking to @sisuresh, this is trickier than I thought. Making all issuer-created claimable balances clawback enabled breaks backwards compatibility (as soon as the protocol 16 upgrade happens, this behavior will change). Making it depend on the account flag of the issuer isn't great, because the issuer might want to be able to clawback their own claimable balances without having clawback enabled for all new trust lines.

There are some ways to workaround this. Issuers could do the "add themselves as a claimant" method that we have discussed before. Alternatively, issuers could send claimable balances from a distribution account that they control which has a clawback enabled trustline.

Nothing here is really great, as far as I'm concerned. What do you think is the best approach @leighmcculloch @sisuresh?

Reply:
> Making all issuer-created claimable balances clawback enabled breaks backwards compatibility

Does this occur because an issuer account has no trustline from which to copy the clawback enabled state? That's interesting, I hadn't considered that.

I think we should discuss this problem on the mailing list. I cross-posted the problem here:
https://groups.google.com/g/stellar-dev/c/hPhkXhrl5-Y/m/Eg6VwKiCAQAJ

Reply:
I update the implementation so the issuer can clawback a claimable balance it creates if `AUTH_CLAWBACK_ENABLED_FLAG` is set on the issuer at the time of balance creation.

Reply:
I'm updating the CAP in https://github.com/stellar/stellar-protocol/pull/857.